lib/core/viewer/components/viewer.component.ts

```diff
@@ -416,6 +419,10 @@ export class ViewerComponent implements OnChanges, OnInit, OnDestroy {
             }
         }
 
+        if (this.viewerType === 'media') {
+            this.mediaTracks = await this.viewUtils.generateMediaTracks(nodeData.id);
```

Comment:
should this thing be in adf-media-player ?

Reply:
We can do it here or in the media player. I decided to do it here and keep media without any logic to follow the same approach that has been done with the source url, that is not calculated in media player but here in the viewer.

I can do it as you prefer, @eromano media or viewer?

Reply:
to me looks a media player concern and you are right in this class there are other concern to move